### PR TITLE
Bluetooth: controller: fixing error re. conn param req ack/req

### PR DESCRIPTION
**Pull request by erbr-ot**
During testing towards Samsung S9, we have seen instances where we as peripheral would send a PHY request during an ongoing conn param req, and it was found that this was due to a bug in handling of conn param req on our part.

llcp.conn_param.ack was reset on tx of conn param rsp, however this
should only be done once the expected conn update ind is received
and sets the cu.ack flag to indicate cu procedure 'takes over'

### CODE REVIEW

**Review by cvinayak — CHANGES_REQUESTED** [review]

**Comment by erbr-ot** [comment]
Implemented suggested changes. Pending test on target.

File: subsys/bluetooth/controller/ll_sw/ull_conn.c [hunk 1]
```diff
@@ -4335,7 +4332,8 @@ static uint8_t conn_upd_recv(struct ll_conn *conn, memq_link_t *link,
 
 #if defined(CONFIG_BT_CTLR_CONN_PARAM_REQ)
 	if ((conn->llcp_conn_param.req != conn->llcp_conn_param.ack) &&
-	    (conn->llcp_conn_param.state == LLCP_CPR_STATE_RSP_WAIT)) {
+	    ((conn->llcp_conn_param.state == LLCP_CPR_STATE_RSP_WAIT) ||
+	     (conn->llcp_conn_param.state == LLCP_CPR_STATE_RSP))) {
```
[the remark this answers is not in the document]
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_conn.c]: My requested change was to only replace `LLCP_CPR_STATE_RSP`. Same for the next one. Let me fix and push.

File: subsys/bluetooth/controller/ll_sw/ull_conn_types.h [hunk 2]
```diff
@@ -230,6 +230,7 @@ struct ll_conn {
 			LLCP_CPR_STATE_APP_WAIT,
 			LLCP_CPR_STATE_RSP_WAIT,
 			LLCP_CPR_STATE_UPD,
+			LLCP_CPR_STATE_UPD_WAIT,
 			LLCP_CPR_STATE_OFFS_REQ,
 			LLCP_CPR_STATE_OFFS_RDY,
 		} state:3 __packed;
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_conn_types.h:236]:
```suggestion
		} state:4 __packed;
```

File: subsys/bluetooth/controller/ll_sw/ull_conn_types.h [hunk 3]
```diff
@@ -230,6 +230,7 @@ struct ll_conn {
 			LLCP_CPR_STATE_APP_WAIT,
 			LLCP_CPR_STATE_RSP_WAIT,
 			LLCP_CPR_STATE_UPD,
+			LLCP_CPR_STATE_UPD_WAIT,
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_conn_types.h:233]:
```suggestion
			LLCP_CPR_STATE_UPD_WAIT,
			LLCP_CPR_STATE_UPD,
```

**Review by cvinayak — APPROVED** [review]

**Review by thoh-ot — APPROVED** [review]

**Pull request merged**